### PR TITLE
Add design system

### PR DESCRIPTION
**Pull request by wmontwe**
This adds a baseline design system with text and button atoms.

### CODE REVIEW

**Review by cketti — COMMENTED** [review]
Do we really need the `Tb` prefix on all the composables? I think it would be nicer if we go rid of them and named our components like the material components. I suggest we then use import aliases to prefix material composables with `Material`.

**Comment by wmontwe** [comment]
> Do we really need the `Tb` prefix on all the composables? I think it would be nicer if we go rid of them and named our components like the material components. I suggest we then use import aliases to prefix material composables with `Material`.

When developing with Compose, Android Studio tends to allways suggest the original Material components instead of the refined ones. This leads to always having to look for the right version beeing imported. In a lot of cases differences are also subtle and easily overlooked. Also without a namespace the only way to figure out which components are actually different to the ones provided by Compose, you have to know them by hearth or look the up in the corresponding package.

**Comment by cketti** [comment]
If we have our own design system we probably shouldn't use Material components outside of the designsystem module anyway. In that case we should be able to change the compose Gradle modules to not add the material library as a dependency.

**Review by cketti — COMMENTED** [review]
`thunderbird.app.android.compose.gradle.kts` and `thunderbird.library.android.compose.gradle.kts` still contain a dependency on `androidx.compose.material:material`. We probably want to change this to `androidx.compose.foundation:foundation`.

Doing the following would clean up the history quite a bit:
1. Squash 781ca5ab9384c0dc685d3c3a13a257684db6ca8e, a46843732619ea6d95128b3711867634e62c2b48, and 09f4f0a42b5f040153125f8d9bc872977c4ae270
2. Squash 781ca5ab9384c0dc685d3c3a13a257684db6ca8e and dfcf72250b046214a01955928394b6a19b1f7caa

**Comment by wmontwe** [comment]
> Doing the following would clean up the history quite a bit:
> 
>     1. Squash [781ca5a](https://github.com/thundernest/k-9/commit/781ca5ab9384c0dc685d3c3a13a257684db6ca8e), [a468437](https://github.com/thundernest/k-9/commit/a46843732619ea6d95128b3711867634e62c2b48), and [09f4f0a](https://github.com/thundernest/k-9/commit/09f4f0a42b5f040153125f8d9bc872977c4ae270)
> 
>     2. Squash [781ca5a](https://github.com/thundernest/k-9/commit/781ca5ab9384c0dc685d3c3a13a257684db6ca8e) and [dfcf722](https://github.com/thundernest/k-9/commit/dfcf72250b046214a01955928394b6a19b1f7caa)

I'm not a huge fan of rewriting pull-request history, as you loose context of what was there and has been changed during a review process. But anyway, I could of course rewrite the history, even if it was already part of a review.

> `thunderbird.app.android.compose.gradle.kts` and `thunderbird.library.android.compose.gradle.kts` still contain a dependency on `androidx.compose.material:material`. We probably want to change this to `androidx.compose.foundation:foundation`.

As I already have a huge pile of code going through a constant history rewriting process, this change slipped into one of the follow up changes. I'll fix that.

File: core/ui/compose/demo/src/main/java/app/k9mail/core/ui/compose/demo/MainView.kt [hunk 1]
```diff
@@ -39,23 +59,57 @@ fun MainView(
...
 }
 
+@Composable
+private fun ButtonContent() {
+    Column {
+        TextHeadline6(text = "Buttons")
+
+        Button(text = "TbButton", onClick = { /*TODO*/ })
+        ButtonOutlined(text = "TbButtonOutlined", onClick = { /*TODO*/ })
+        ButtonText(text = "TbButtonText", onClick = { /*TODO*/ })
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply core/ui/compose/demo/src/main/java/app/k9mail/core/ui/compose/demo/MainView.kt:88]: I'll fix that. Slipped into one of the follow up changes.

**Comment by cketti** [comment]
> I'm not a huge fan of rewriting pull-request history, as you loose context of what was there and has been changed during a review process. But anyway, I could of course rewrite the history, even if it was already part of a review.

Most of the things changed/fixed during a code review are not interesting in a historical context. So I'd like to avoid merging fixup commits into `main`. Personally, I find it easier to rewrite the pull request history right away. And since GitHub displays diffs for force-pushes, it's relatively easy for a reviewer to find out what has changed. 
But I also think it's fine to add fixup commits to a pull request and then rewrite the history before merging the branch. We should probably add a label to such pull requests so the rewrite step at the end isn't forgotten.

**Comment by wmontwe** [comment]
A label will work

**Review by cketti — APPROVED** [review]
:+1:

**Pull request merged**